packages/kit/types/ambient.d.ts

```diff
@@ -243,7 +243,8 @@ declare module '$app/navigation' {
 
 	/**
 	 * A navigation interceptor that triggers before we navigate to a new URL, whether by clicking a link, calling `goto(...)`, or using the browser back/forward controls.
-	 * Calling `cancel()` will prevent the navigation from completing.
+	 * Calling `cancel()` will prevent the navigation from completing. If the navigation would have directly unloaded the current page, calling `cancel` will trigger the native
+	 * browser unload confirmation dialog. This is the case for type `unload` and for `popstate` if the next url is not part of the app.
```

Comment:
is this right? `popstate` would _only_ fire for internal navigations, surely? if clicking the back button would cause us to go back to a different document then type would be `unload`. 

```suggestion
	 * Calling `cancel()` will prevent the navigation from completing. If the navigation would have directly unloaded the current page, calling `cancel` will trigger the native
	 * browser unload confirmation dialog. In these cases, `navigation.willUnload` is `true`.
```